content/livesync/models/models.textile

```diff
@@ -0,0 +1,291 @@
+
+When using the event buffer, change events are de-duplicated and ordered according to their "@message.ID@":/api/realtime-sdk/messages#id, which corresponds to the change event's "@sequenceID@":#sequence.
+
+By default, the events in the buffer are ordered numerically if the @message.ID@ can be coerced to a number.
+Otherwise, events will be ordered lexicographically by their @message.ID@.
+
+You can specify a custom ordering based on any part of the message via the @eventBufferOptions.eventOrderer@ "@ClientOption@":/livesync/models#options.
+
+h2(#optimistic). Optimistic updates
+
+The Models SDK supports optimistic updates, a feature that enables you to immediately render changes to your data model before the backend confirms these changes, making updates appear instantaneous.
+
+Optimistic events are used to make local, provisional changes to your data, anticipating that your backend will eventually confirm or reject these changes. The optimistic event will be processed by your "merge function":#merge and included in the local model state optimistically.
+
+Use the @model.optimistic()@ method to apply an optimistic update to your model.
+
+1. Call @model.optimistic()@ on your model with the optimistic event.
+2. Apply the corresponding change to your backend.
+3. Await the confirmation of the optimistic update from the backend, or optionally cancel the optimistic update.
+
+The @model.optimistic()@ function returns a promise resolving to two values:
+
+* A confirmation promise that resolves when the backend confirms or rejects the optimistic update.
+* A function to explicitly cancel the optimistic update.
+
+The following is an example of optimistically applying changes to a model:
+
+```[javascript]
+// optimistically apply the changes to the model
+const [confirmation, cancel] = await model.optimistic({
+    mutationID: 'my-mutation-id',
+    name: 'addComment',
+    data: 'New comment!',
+});
+
+try {
+    // apply the changes in your backend
+    await updatePost('my-mutation-id', 'New comment!');
+    // wait for the optimistic event to be confirmed
+    await confirmation;
+} catch (err) {
+    // something went wrong, cancel the optimistic update
+    cancel();
+}
+```
+
+Be aware that updating the model state in your database happens entirely via your backend endpoint.
+
+h3(#mutation). MutationID
+
+The Models SDK needs a way to know when a change event received from the backend corresponds to an optimistic event that has been applied locally.
+
+To do this, your clients must define a unique @mutationID@ on the optimistic event. This ID can be any string, but typically this will be a UUID generated from a library, such as "uuid"https://www.npmjs.com/package/uuid. The @mutationID@ should then be passed to your backend and included on the confirmation event it writes to the "outbox":/livesync/connector/database#outbox, for example:
```

Comment:
The link on the `uuid` is broken, I think missing a colon

Reply:
Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)